src/error.rs

```diff
@@ -10,15 +10,15 @@ pub type Result<T> = std::result::Result<T, Error>;
 
 /// The Errors that may occur when processing a `Request`.
 pub struct Error {
-    inner: Box<Inner>,
+    pub(crate) inner: Box<Inner>,
```

Comment:
I'd leave these changes out, personally. It's fine to just test the publicly observable properties of the `Error`.

Reply:
What publicly-observable properties would you recommend verifying? I tried to compare the properties that seemed unlikely to change (Kind, Url) instead of formatting the error and doing a string comparison, because the latter seemed like it would be unnecessarily fragile to changes in the error string.

Reply:
We can test `is_builder()`, and `url()`.

Reply:
Oh, thanks! I had missed those.